### DISCUSSION TITLE
Teleport helm charts: how to reload a changed configmap?

### DISCUSSION THREAD

**Discussion by urbaman** [gravitational/teleport#29299] [Q&A] [answered]
Hi,

I deployed both the teleport-cluster and the teleport-kube-agent helm charts, and made them work.
Now, I'd like to edit the teleport-kube-agent configmap to add more resources (app, ...).
I think the configmap reloading goes fine, but the teleport instance inside the pod do not reload the new config, so not registering the new/edited resourses.
Is there a way to reload the teleport instance without deleting/evicting the pod? I didn't find any teleport/tsh/tctl command to reload a config.
The same could be said for the proxy/auth pods of the teleport-cluster chart.

TIP: in a "normal" installation through systamd, systemctl reload works reloading the new config.

**Comment by webvictim** [comment, marked as answer]
The expected Helm flow for adding new apps to `teleport-kube-agent` is to update the `apps: []` list in the Helm values (https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-kube-agent/values.yaml#L43-L47) and then run `helm upgrade` to deploy a new pod version with the changes.

Editing the `ConfigMap` in place isn't officially supported and is likely to cause you issues. If you really want to, though, try `kubectl -n <namespace> rollout restart deploy/<release-name>`
